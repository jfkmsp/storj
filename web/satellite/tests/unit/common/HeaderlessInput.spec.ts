// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import HeaderlessInput from '@/components/common/HeaderlessInput.vue';

import { mount, shallowMount } from '@vue/test-utils';

describe('HeaderlessInput.vue', () => {
    it('renders correctly with default props', () => {

        const wrapper = shallowMount(HeaderlessInput);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with size props', () => {
        const placeholder = 'test';
        const width = '30px';
        const height = '20px';

        const wrapper = shallowMount(HeaderlessInput, {
            propsData: {placeholder, width, height},
        });

        const el = wrapper.find('input').element as HTMLElement;
        expect(el.style.width).toMatch(width);
        expect(el.style.height).toMatch(height);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with isPassword prop', () => {
        const wrapper = mount(HeaderlessInput, {
            propsData: {isPassword: true},
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('emit setData on input correctly', () => {
        const testData = 'testData';

        const wrapper = mount(HeaderlessInput);

        wrapper.find('input').trigger('input');

        let emittedSetData = wrapper.emitted('setData');
        if (emittedSetData) expect(emittedSetData.length).toEqual(1);

        wrapper.vm.$emit('setData', testData);

        emittedSetData = wrapper.emitted('setData');
        if (emittedSetData) expect(emittedSetData[1][0]).toEqual(testData);
    });

});
