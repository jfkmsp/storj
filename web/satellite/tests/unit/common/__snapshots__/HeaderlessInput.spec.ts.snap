// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderlessInput.vue renders correctly with default props 1`] = `
<div aria-roledescription="input-container" class="input-wrap">
  <div class="label-container">
    <!---->
    <!---->
    <!---->
  </div> <input placeholder="default" type="text" class="headerless-input" style="width: 100%; height: 48px;">
  <!---->
  <!---->
  <!---->
  <!---->
</div>
`;

exports[`HeaderlessInput.vue renders correctly with isPassword prop 1`] = `
<div aria-roledescription="input-container" class="input-wrap">
  <div class="label-container">
    <!---->
    <!---->
    <!---->
  </div> <input placeholder="default" type="password" class="headerless-input password" style="width: 100%; height: 48px;">
  <!---->
  <!----> <svg class="input-wrap__image"></svg>
  <!---->
</div>
`;

exports[`HeaderlessInput.vue renders correctly with size props 1`] = `
<div aria-roledescription="input-container" class="input-wrap">
  <div class="label-container">
    <!---->
    <!---->
    <!---->
  </div> <input placeholder="test" type="text" class="headerless-input" style="width: 30px; height: 20px;">
  <!---->
  <!---->
  <!---->
  <!---->
</div>
`;
